### PR TITLE
Sonar/warnings (#992)

### PR DESCRIPTION
**Pull request by marvin-bitterlich**
* Fix Blockers

* first draft for translation constants

* fix formatting

* fix some constants and other issues

* formatting

* fix some ternaries

* sonar fixes

* add suppress

* add doc files

* reorder

* wtf java

* last fixes

* remove unused

merge sucked I hope that worked…

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java [hunk 1]
```diff
@@ -106,6 +112,11 @@ public InventoryCitizen(final String title, final boolean localeEnabled)
         {
             customName = title;
         }
+        this.allInventories = new ArrayList<>();
+        this.allInventories.add(this.mainInventory);
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java]: ?

File: src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java [hunk 2]
```diff
@@ -57,10 +58,10 @@ public BlockBarrel()
     }
 
     //todo: register block with new method
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java:60]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [BuildingFarmer.java#L499](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java#L499): Reduce this class from 86 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L154](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L154): Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L57](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/structures/helpers/Structure.java#L57): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 36 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BlockBarrel.java#L68](https://github.com/Minecolonies/minecolonies/blob/18b04138711678b935a2265805b38ba68d39bc1d/src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java#L68): Define the locale to be used in this String operation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [BuildingFarmer.java#L499](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java#L499): Reduce this class from 86 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L154](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L154): Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L57](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/structures/helpers/Structure.java#L57): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 36 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BlockBarrel.java#L68](https://github.com/Minecolonies/minecolonies/blob/426858cb56206ab8e276e2f65751e096e999c347/src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java#L68): Define the locale to be used in this String operation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/995/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**